presto-orc/src/main/java/com/facebook/presto/orc/reader/MapStreamReader.java

```diff
@@ -128,8 +126,8 @@ public Block readBlock(Type type)
         Type valueType = mapType.getValueType();
 
         int entryCount = 0;
-        for (int length : lengthVector) {
-            entryCount += length;
+        for (int i = 0; i < offsetVector.length - 1; i++) {
+            entryCount += offsetVector[i];
```

Comment:
nit: Add a comment explain here `offsetVector` still stores length (instead of offset). Otherwise someone who happen to read this code will be confused without understanding the optimization.